_posts/2018-02-28-Update-to-Java11.md

```diff
@@ -0,0 +1,124 @@
+
+## Whip out the Docker
+
+To try out the early access build of Java 11 you need a nice Linux base image, and some JDK install magic on top of that. Here's an example Dockerfile:
+
+```
+FROM centos
+
+ENV LANG C.UTF-8
+ENV JAVA_PACKAGE_URL https://download.java.net/java/early_access/jdk11/2/GPL/openjdk-11-ea+2_linux-x64_bin.tar.gz
+
+# Upgrading system
+RUN yum -y upgrade
+RUN yum -y install wget
+
+WORKDIR /opt
+
+RUN wget --no-cookies --no-check-certificate \
+  $JAVA_PACKAGE_URL -O /opt/jdk11.tar.gz \
+  && tar -zxvf jdk11.tar.gz \
+  && rm jdk11.tar.gz
+
+ENV JAVA_HOME /opt/jdk-11
+ENV PATH "$PATH:$JAVA_HOME/bin"
+
+RUN useradd -ms /bin/bash javauser
+USER javauser
+WORKDIR /home/javauser
+
+CMD jshell
+```
+
+It's very crude but gets the thing done for now. If you are impatient, you can also use our this image directly from Docker hub. I do not guarantee long support though, I'm sure better images will start emergin anytime now.
+
+```
+docker run -it solita/jdk11-ea
+```
+
+Got it? Go play!
+
+## What's new?
+
+Not so much is new. Java 10 and 11 are still moving targets, and lots will change as well. But some things that we know are already out there.
+
+Java 10 is to be released any moment now. Java 9 public updates are also destined to end soon, so you should update to Java 10 soon if you like those updates. Java 10 is going to be another short term support update, with Java 11 being a long term support update, like Java 8.
+
+![Some of the JDK 10/11 goodies](/img/java11/jshell_inference.png)
+
+Java 10 will have type inference for local variable type (YAY!), so your code will look like this:
+
+```Java
+// Upcoming Java 10 code
+var productList = new ArrayList<String>();
+productList.add("Macbook Pro");
+var productStream = productList.stream();
+productStream.forEach(System.out::println);
+```
+
+Note a few things. This requires you to define the generic typing on the right side, so it can be inferred. No diamond operator for you. Secondly, there's no 'val', only var. Meaning, it's just a normal mutable variable declaration, with a bit syntactic sugar. Still nice. You can test it immediately with Java 10/11 early access build, and soon with Java 10 GA build.
+
+It will not work for method/constructor parameters or return types, member variables/fields, or catch formals. Only for logic within methods, including loop indexes.
+
+Of course, we still have all Java 9 goodies such as new Stream functions:
+
+```
+jshell> IntStream.range(0,10).takeWhile(x -> x < 5).forEach(System.out::println)
+0
+1
+2
+3
+4
+```
+
+If you update to any of these Java versions, 9, 10, 11, you get access to JShell and goodies like that. JShell that was introduced in Java 9, starts up now significantly faster, making it much better REPL experience for those who value it.
```

Comment:
"Java versions (9, 10, or 11)"

"goodies like that" -> "other goodies" maybe`

"JShell, which was introduced… now starts up… making it **a** much better"